Incompatible heapless version used to initialize embassy-net static configuration

### Summary

The version of embassy-net we use is using heapless v0.8.0, using heapless v0.9.1 breaks the type compatibility in `src/ariel-os-embassy/src/net.rs:249:26`. 

To reproduce the issue, you can run

```
laze -C examples/tcp-client/ build -b nrf52840dk -s ipv6   
```

### Relevant log output

```shell
   Compiling ariel-os-embassy v0.3.0 (/tmp/4133/ariel-os/src/ariel-os-embassy)
error[E0308]: mismatched types
   --> src/ariel-os-embassy/src/net.rs:249:26
    |
249 |             dns_servers: heapless::Vec::new(),
    |                          ^^^^^^^^^^^^^^^^^^^^ expected `Vec<Ipv6Address, 3>`, found `VecInner<_, _, ...>`
    |
    = note: expected struct `heapless::vec::Vec<Ipv6Addr, 3>`
               found struct `VecInner<_, _, vec::storage::VecStorageInner<[MaybeUninit<_>; _]>>`

```

### Hardware/ Setup

_No response_

### Possible Solution

_No response_

### Would you like to work on fixing this bug?

Maybe

--- a/src/ariel-os-embassy/Cargo.toml
+++ b/src/ariel-os-embassy/Cargo.toml
@@ -42,7 +42,6 @@ ariel-os-storage = { workspace = true, optional = true }
 ariel-os-threads = { path = "../ariel-os-threads", optional = true }
 ariel-os-utils = { workspace = true }
 
-heapless = "0.9.1"
 trouble-host = { workspace = true, optional = true }
 usbd-hid = { version = "0.8.2", optional = true }
 

--- a/src/ariel-os-embassy/src/net.rs
+++ b/src/ariel-os-embassy/src/net.rs
@@ -212,7 +212,11 @@ fn __ariel_os_network_config() -> embassy_net::Config {
 
             config.ipv4 = embassy_net::ConfigV4::Static(embassy_net::StaticConfigV4 {
                 address: embassy_net::Ipv4Cidr::new(ipaddr, PREFIX_LEN),
-                dns_servers: heapless::Vec::new(),
+                #[expect(
+                    clippy::default_trait_access,
+                    reason = "This allows us to not import heapless (and not worry about its version)."
+                )]
+                dns_servers: Default::default(),
                 gateway: Some(gw_addr),
             });
         } else if #[cfg(feature = "dhcpv4")] {
@@ -246,7 +250,11 @@ fn __ariel_os_network_config() -> embassy_net::Config {
 
         config.ipv6 = embassy_net::ConfigV6::Static(embassy_net::StaticConfigV6 {
             address: embassy_net::Ipv6Cidr::new(ipaddr, PREFIX_LEN),
-            dns_servers: heapless::Vec::new(),
+            #[expect(
+                clippy::default_trait_access,
+                reason = "This allows us to not import heapless (and not worry about its version)."
+            )]
+            dns_servers: Default::default(),
             gateway: Some(gw_addr),
         });
     }

